aptos-move/framework/aptos-stdlib/sources/cryptography/ristretto255.move

```diff
@@ -0,0 +1,1271 @@
+/// This module contains functions for Ristretto255 curve arithmetic, assuming addition as the group operation.
+///
+/// The order of the Ristretto255 elliptic curve group is $\ell = 2^252 + 27742317777372353535851937790883648493$, same
+/// as the order of the prime-order subgroup of Curve25519.
+///
+/// This module provides two structs for encoding Ristretto elliptic curves to the developer:
+///
+///  - First, a 32-byte-sized CompressedRistretto struct, which is used to persist points in storage.
+///
+///  - Second, a larger, in-memory, RistrettoPoint struct, which is decompressable from a CompressedRistretto struct. This
+/// larger struct can be used for fast arithmetic operations (additions, multiplications, etc.). The results can be saved
+/// back into storage by compressing RistrettoPoint structs back to CompressedRistretto structs.
+///
+/// This module also provides a Scalar struct for persisting scalars in storage and doing fast arithmetic on them.
+///
+/// One invariant maintained by this module is that all CompressedRistretto structs store a canonically-encoded point,
+/// which can always be decompressed into a valid point on the curve as a RistrettoPoint struct. Unfortunately, due to
+/// limitations in our underlying curve25519-dalek elliptic curve library, this decompression will unnecessarily verify
+/// the validity of the point and thus slightly decrease performance.
+///
+/// Similarly, all Scalar structs store a canonically-encoded scalar, which can always be safely operated on using
+/// arithmetic operations.
+///
+/// In the future, we might support additional features:
+///
+/// * For scalars:
+///    - batch_invert()
+///
+///  * For points:
+///    - double()
+///      + The challenge is that curve25519-dalek does NOT export double for Ristretto points (nor for Edwards)
+///
+///    - double_and_compress_batch()
+///
+///    - fixed-base, variable-time via optional_mixed_multiscalar_mul() in VartimePrecomputedMultiscalarMul
+///      + This would require a storage-friendly RistrettoBasepointTable and an in-memory variant of it too
+///      + Similar to the CompressedRistretto and RistrettoPoint structs in this module
+///      + The challenge is that curve25519-dalek's RistrettoBasepointTable is not serializable
+
+module aptos_std::ristretto255 {
+    use std::option::Option;
+
+    #[test_only]
+    use std::option;
+
+    //
+    // Constants
+    //
+
+    /// The order of the Ristretto255 group and its scalar field, in little-endian.
+    const ORDER_ELL : vector<u8> = vector[
+        0xed, 0xd3, 0xf5, 0x5c, 0x1a, 0x63, 0x12, 0x58,
+        0xd6, 0x9c, 0xf7, 0xa2, 0xde, 0xf9, 0xde, 0x14,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00,
+        0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x00, 0x10,
+    ];
```

Comment:
you can define this as x"edd3f55c1a63..."

Reply:
Yes, I will change all of these: they are copied & pasted examples from `curve25519-dalek` code.